Fix types in MapView.web

pnpm run typecheck --filter functions
で、以下のエラーが出ていますので、型が合うように
nanicore-audio-guide/app-expo/components/MapView.web.tsx
を修正してください。


> app-expo@1.1.0 typecheck /root/workspace/nanicore-audio-guide/app-expo
> tsc --noEmit

components/MapView.web.tsx:9:3 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: MarkerProps): Marker', gave the following error.
    Type '((event: MarkerPressEvent) => void) | undefined' is not assignable to type '((e: MapMouseEvent) => void) | undefined'.
      Type '(event: MarkerPressEvent) => void' is not assignable to type '(e: MapMouseEvent) => void'.
        Types of parameters 'event' and 'e' are incompatible.
          Type 'MapMouseEvent' is missing the following properties from type 'NativeSyntheticEvent<{ id: string; action: "marker-press"; coordinate: LatLng; position?: Point | undefined; }>': nativeEvent, currentTarget, target, bubbles, and 11 more.
  Overload 2 of 2, '(props: MarkerProps, context: any): Marker', gave the following error.
    Type '((event: MarkerPressEvent) => void) | undefined' is not assignable to type '((e: MapMouseEvent) => void) | undefined'.
      Type '(event: MarkerPressEvent) => void' is not assignable to type '(e: MapMouseEvent) => void'.
        Types of parameters 'event' and 'e' are incompatible.
          Type 'MapMouseEvent' is missing the following properties from type 'NativeSyntheticEvent<{ id: string; action: "marker-press"; coordinate: LatLng; position?: Point | undefined; }>': nativeEvent, currentTarget, target, bubbles, and 11 more.

9   onClick={onPress}
    ~~~~~~~

  ../node_modules/@react-google-maps/api/dist/index.d.ts:315:5
    315     onClick?: ((e: google.maps.MapMouseEvent) => void) | undefined;
            ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Marker> & Readonly<MarkerProps>'
  ../node_modules/@react-google-maps/api/dist/index.d.ts:315:5
    315     onClick?: ((e: google.maps.MapMouseEvent) => void) | undefined;
            ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Marker> & Readonly<MarkerProps>'

components/MapView.web.tsx:33:7 - error TS2554: Expected 2 arguments, but got 1.

33       onRegionChangeComplete({
         ~~~~~~~~~~~~~~~~~~~~~~

  ../node_modules/react-native-maps/lib/MapView.d.ts:334:47
    334     onRegionChangeComplete?: (region: Region, details: Details) => void;
                                                      ~~~~~~~~~~~~~~~~
    An argument for 'details' was not provided.

components/MapView.web.tsx:36:23 - error TS18048: 'region' is possibly 'undefined'.

36        latitudeDelta: region.latitudeDelta,
                         ~~~~~~

components/MapView.web.tsx:37:24 - error TS18048: 'region' is possibly 'undefined'.

37        longitudeDelta: region.longitudeDelta,
                          ~~~~~~

components/MapView.web.tsx:48:21 - error TS18048: 'region' is possibly 'undefined'.

48      center={{ lat: region.latitude, lng: region.longitude }}
                       ~~~~~~

components/MapView.web.tsx:48:43 - error TS18048: 'region' is possibly 'undefined'.

48      center={{ lat: region.latitude, lng: region.longitude }}
                                             ~~~~~~

components/MapView.web.tsx:50:58 - error TS2698: Spread types may only be created from object types.

50      mapContainerStyle={{ width: "100%", height: "100%", ...style }}
                                                            ~~~~~~~~

components/MapView.web.tsx:51:6 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(props: GoogleMapProps): GoogleMap', gave the following error.
    Type '((event: MapPressEvent) => void) | undefined' is not assignable to type '((e: MapMouseEvent) => void) | undefined'.
      Type '(event: MapPressEvent) => void' is not assignable to type '(e: MapMouseEvent) => void'.
        Types of parameters 'event' and 'e' are incompatible.
          Type 'MapMouseEvent' is missing the following properties from type 'NativeSyntheticEvent<{ coordinate: LatLng; position: Point; } & { action?: "marker-press" | "press" | undefined; }>': nativeEvent, currentTarget, target, bubbles, and 11 more.
  Overload 2 of 2, '(props: GoogleMapProps, context: any): GoogleMap', gave the following error.
    Type '((event: MapPressEvent) => void) | undefined' is not assignable to type '((e: MapMouseEvent) => void) | undefined'.
      Type '(event: MapPressEvent) => void' is not assignable to type '(e: MapMouseEvent) => void'.
        Types of parameters 'event' and 'e' are incompatible.
          Type 'MapMouseEvent' is missing the following properties from type 'NativeSyntheticEvent<{ coordinate: LatLng; position: Point; } & { action?: "marker-press" | "press" | undefined; }>': nativeEvent, currentTarget, target, bubbles, and 11 more.

51      onClick={onPress}
        ~~~~~~~

  ../node_modules/@react-google-maps/api/dist/index.d.ts:36:5
    36     onClick?: ((e: google.maps.MapMouseEvent) => void) | undefined;
           ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<GoogleMap> & Readonly<GoogleMapProps>'
  ../node_modules/@react-google-maps/api/dist/index.d.ts:36:5
    36     onClick?: ((e: google.maps.MapMouseEvent) => void) | undefined;
           ~~~~~~~
    The expected type comes from property 'onClick' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<GoogleMap> & Readonly<GoogleMapProps>'


Found 8 errors in the same file, starting at: components/MapView.web.tsx:9

--- a/app-expo/components/MapView.web.tsx
+++ b/app-expo/components/MapView.web.tsx
@@ -1,142 +1,137 @@
 import React from "react";
-import {
-    GoogleMap,
-    Marker as GoogleMarker,
-    LoadScript,
-} from "@react-google-maps/api";
+import { GoogleMap, Marker as GoogleMarker, LoadScript } from "@react-google-maps/api";
 import type { MapViewProps, MarkerProps } from "./MapView";
 
 // React Native event types used by MapView and Marker
 interface LatLng {
-    latitude: number;
-    longitude: number;
+	latitude: number;
+	longitude: number;
 }
 
 interface Point {
-    x: number;
-    y: number;
+	x: number;
+	y: number;
 }
 
 interface MarkerPressEvent {
-    nativeEvent: {
-        id: string;
-        action: "marker-press";
-        coordinate: LatLng;
-        position?: Point;
-    };
+	nativeEvent: {
+		id: string;
+		action: "marker-press";
+		coordinate: LatLng;
+		position?: Point;
+	};
 }
 
 interface MapPressEvent {
-    nativeEvent: {
-        coordinate: LatLng;
-        position: Point;
-        action?: "marker-press" | "press";
-    };
+	nativeEvent: {
+		coordinate: LatLng;
+		position: Point;
+		action?: "marker-press" | "press";
+	};
 }
 
 interface RegionChangeDetails {
-    isGesture: boolean;
+	isGesture: boolean;
 }
 
 export const Marker: React.FC<MarkerProps> = ({ coordinate, title, onPress, testID }) => {
-    const handleClick = React.useCallback(
-        (e: google.maps.MapMouseEvent) => {
-            if (!onPress || !e.latLng) return;
-            const event: MarkerPressEvent = {
-                nativeEvent: {
-                    id: testID ?? "",
-                    action: "marker-press",
-                    coordinate: {
-                        latitude: e.latLng.lat(),
-                        longitude: e.latLng.lng(),
-                    },
-                },
-            };
-            onPress(event);
-        },
-        [onPress, testID]
-    );
+	const handleClick = React.useCallback(
+		(e: google.maps.MapMouseEvent) => {
+			if (!onPress || !e.latLng) return;
+			const event: MarkerPressEvent = {
+				nativeEvent: {
+					id: testID ?? "",
+					action: "marker-press",
+					coordinate: {
+						latitude: e.latLng.lat(),
+						longitude: e.latLng.lng(),
+					},
+				},
+			};
+			onPress(event);
+		},
+		[onPress, testID],
+	);
 
-    return (
-        <GoogleMarker
-            position={{ lat: coordinate.latitude, lng: coordinate.longitude }}
-            title={title}
-            onClick={handleClick}
-            options={{}}
-        />
-    );
+	return (
+		<GoogleMarker
+			position={{ lat: coordinate.latitude, lng: coordinate.longitude }}
+			title={title}
+			onClick={handleClick}
+			options={{}}
+		/>
+	);
 };
 
 const MapView = React.forwardRef<google.maps.Map | null, MapViewProps>(
-    ({ style, region, onRegionChangeComplete, onPress, children }, ref) => {
-        const handleLoad = React.useCallback(
-            (map: google.maps.Map) => {
-                if (typeof ref === "function") {
-                    ref(map);
-                } else if (ref) {
-                    (ref as React.MutableRefObject<google.maps.Map | null>).current = map;
-                }
-            },
-            [ref]
-        );
+	({ style, region, onRegionChangeComplete, onPress, children }, ref) => {
+		const handleLoad = React.useCallback(
+			(map: google.maps.Map) => {
+				if (typeof ref === "function") {
+					ref(map);
+				} else if (ref) {
+					(ref as React.MutableRefObject<google.maps.Map | null>).current = map;
+				}
+			},
+			[ref],
+		);
 
-        const handleIdle = React.useCallback(() => {
-            if (onRegionChangeComplete && ref && typeof ref !== "function") {
-                const map = (ref as React.MutableRefObject<google.maps.Map | null>).current;
-                if (map) {
-                    const center = map.getCenter();
-                    if (center) {
-                        const eventRegion = {
-                            latitude: center.lat(),
-                            longitude: center.lng(),
-                            latitudeDelta: region?.latitudeDelta ?? 0,
-                            longitudeDelta: region?.longitudeDelta ?? 0,
-                        };
-                        const details: RegionChangeDetails = { isGesture: false };
-                        onRegionChangeComplete(eventRegion, details);
-                    }
-                }
-            }
-        }, [onRegionChangeComplete, region, ref]);
+		const handleIdle = React.useCallback(() => {
+			if (onRegionChangeComplete && ref && typeof ref !== "function") {
+				const map = (ref as React.MutableRefObject<google.maps.Map | null>).current;
+				if (map) {
+					const center = map.getCenter();
+					if (center) {
+						const eventRegion = {
+							latitude: center.lat(),
+							longitude: center.lng(),
+							latitudeDelta: region?.latitudeDelta ?? 0,
+							longitudeDelta: region?.longitudeDelta ?? 0,
+						};
+						const details: RegionChangeDetails = { isGesture: false };
+						onRegionChangeComplete(eventRegion, details);
+					}
+				}
+			}
+		}, [onRegionChangeComplete, region, ref]);
 
-        const handleClick = React.useCallback(
-            (e: google.maps.MapMouseEvent) => {
-                if (!onPress || !e.latLng) return;
-                const pressEvent: MapPressEvent = {
-                    nativeEvent: {
-                        coordinate: {
-                            latitude: e.latLng.lat(),
-                            longitude: e.latLng.lng(),
-                        },
-                        position: { x: 0, y: 0 },
-                    },
-                };
-                onPress(pressEvent);
-            },
-            [onPress]
-        );
+		const handleClick = React.useCallback(
+			(e: google.maps.MapMouseEvent) => {
+				if (!onPress || !e.latLng) return;
+				const pressEvent: MapPressEvent = {
+					nativeEvent: {
+						coordinate: {
+							latitude: e.latLng.lat(),
+							longitude: e.latLng.lng(),
+						},
+						position: { x: 0, y: 0 },
+					},
+				};
+				onPress(pressEvent);
+			},
+			[onPress],
+		);
 
-        const containerStyle: React.CSSProperties = {
-            width: "100%",
-            height: "100%",
-            ...(typeof style === "object" && style !== null ? (style as React.CSSProperties) : {}),
-        };
+		const containerStyle: React.CSSProperties = {
+			width: "100%",
+			height: "100%",
+			...(typeof style === "object" && style !== null ? (style as React.CSSProperties) : {}),
+		};
 
-        return (
-            <LoadScript googleMapsApiKey="">
-                <GoogleMap
-                    onLoad={handleLoad}
-                    center={{ lat: region?.latitude ?? 0, lng: region?.longitude ?? 0 }}
-                    zoom={10}
-                    mapContainerStyle={containerStyle}
-                    onClick={handleClick}
-                    onIdle={handleIdle}
-                >
-                    {children}
-                </GoogleMap>
-            </LoadScript>
-        );
-    }
+		return (
+			<LoadScript googleMapsApiKey="">
+				<GoogleMap
+					onLoad={handleLoad}
+					center={{ lat: region?.latitude ?? 0, lng: region?.longitude ?? 0 }}
+					zoom={10}
+					mapContainerStyle={containerStyle}
+					onClick={handleClick}
+					onIdle={handleIdle}>
+					{children}
+				</GoogleMap>
+			</LoadScript>
+		);
+	},
 );
 
 export default MapView;

--- a/app-expo/components/MapView.web.tsx
+++ b/app-expo/components/MapView.web.tsx
@@ -1,60 +1,142 @@
 import React from "react";
-import { GoogleMap, Marker as GoogleMarker, LoadScript } from "@react-google-maps/api";
+import {
+    GoogleMap,
+    Marker as GoogleMarker,
+    LoadScript,
+} from "@react-google-maps/api";
 import type { MapViewProps, MarkerProps } from "./MapView";
 
-export const Marker: React.FC<MarkerProps> = ({ coordinate, title, onPress, testID }) => (
-	<GoogleMarker
-		position={{ lat: coordinate.latitude, lng: coordinate.longitude }}
-		title={title}
-		onClick={onPress}
-		options={{}}
-	/>
-);
+// React Native event types used by MapView and Marker
+interface LatLng {
+    latitude: number;
+    longitude: number;
+}
+
+interface Point {
+    x: number;
+    y: number;
+}
+
+interface MarkerPressEvent {
+    nativeEvent: {
+        id: string;
+        action: "marker-press";
+        coordinate: LatLng;
+        position?: Point;
+    };
+}
+
+interface MapPressEvent {
+    nativeEvent: {
+        coordinate: LatLng;
+        position: Point;
+        action?: "marker-press" | "press";
+    };
+}
+
+interface RegionChangeDetails {
+    isGesture: boolean;
+}
+
+export const Marker: React.FC<MarkerProps> = ({ coordinate, title, onPress, testID }) => {
+    const handleClick = React.useCallback(
+        (e: google.maps.MapMouseEvent) => {
+            if (!onPress || !e.latLng) return;
+            const event: MarkerPressEvent = {
+                nativeEvent: {
+                    id: testID ?? "",
+                    action: "marker-press",
+                    coordinate: {
+                        latitude: e.latLng.lat(),
+                        longitude: e.latLng.lng(),
+                    },
+                },
+            };
+            onPress(event);
+        },
+        [onPress, testID]
+    );
+
+    return (
+        <GoogleMarker
+            position={{ lat: coordinate.latitude, lng: coordinate.longitude }}
+            title={title}
+            onClick={handleClick}
+            options={{}}
+        />
+    );
+};
 
 const MapView = React.forwardRef<google.maps.Map | null, MapViewProps>(
-	({ style, region, onRegionChangeComplete, onPress, children }, ref) => {
-		const handleLoad = React.useCallback(
-			(map: google.maps.Map) => {
-				if (typeof ref === "function") {
-					ref(map);
-				} else if (ref) {
-					(ref as React.MutableRefObject<google.maps.Map | null>).current = map;
-				}
-			},
-			[ref],
-		);
+    ({ style, region, onRegionChangeComplete, onPress, children }, ref) => {
+        const handleLoad = React.useCallback(
+            (map: google.maps.Map) => {
+                if (typeof ref === "function") {
+                    ref(map);
+                } else if (ref) {
+                    (ref as React.MutableRefObject<google.maps.Map | null>).current = map;
+                }
+            },
+            [ref]
+        );
 
-		const handleIdle = React.useCallback(() => {
-			if (onRegionChangeComplete && ref && typeof ref !== "function") {
-				const map = (ref as React.MutableRefObject<google.maps.Map | null>).current;
-				if (map) {
-					const center = map.getCenter();
-					if (center) {
-						onRegionChangeComplete({
-							latitude: center.lat(),
-							longitude: center.lng(),
-							latitudeDelta: region.latitudeDelta,
-							longitudeDelta: region.longitudeDelta,
-						});
-					}
-				}
-			}
-		}, [onRegionChangeComplete, region, ref]);
+        const handleIdle = React.useCallback(() => {
+            if (onRegionChangeComplete && ref && typeof ref !== "function") {
+                const map = (ref as React.MutableRefObject<google.maps.Map | null>).current;
+                if (map) {
+                    const center = map.getCenter();
+                    if (center) {
+                        const eventRegion = {
+                            latitude: center.lat(),
+                            longitude: center.lng(),
+                            latitudeDelta: region?.latitudeDelta ?? 0,
+                            longitudeDelta: region?.longitudeDelta ?? 0,
+                        };
+                        const details: RegionChangeDetails = { isGesture: false };
+                        onRegionChangeComplete(eventRegion, details);
+                    }
+                }
+            }
+        }, [onRegionChangeComplete, region, ref]);
 
-		return (
-			<LoadScript googleMapsApiKey="">
-				<GoogleMap
-					onLoad={handleLoad}
-					center={{ lat: region.latitude, lng: region.longitude }}
-					zoom={10}
-					mapContainerStyle={{ width: "100%", height: "100%", ...style }}
-					onClick={onPress}
-					onIdle={handleIdle}>
-					{children}
-				</GoogleMap>
-			</LoadScript>
-		);
-	},
+        const handleClick = React.useCallback(
+            (e: google.maps.MapMouseEvent) => {
+                if (!onPress || !e.latLng) return;
+                const pressEvent: MapPressEvent = {
+                    nativeEvent: {
+                        coordinate: {
+                            latitude: e.latLng.lat(),
+                            longitude: e.latLng.lng(),
+                        },
+                        position: { x: 0, y: 0 },
+                    },
+                };
+                onPress(pressEvent);
+            },
+            [onPress]
+        );
+
+        const containerStyle: React.CSSProperties = {
+            width: "100%",
+            height: "100%",
+            ...(typeof style === "object" && style !== null ? (style as React.CSSProperties) : {}),
+        };
+
+        return (
+            <LoadScript googleMapsApiKey="">
+                <GoogleMap
+                    onLoad={handleLoad}
+                    center={{ lat: region?.latitude ?? 0, lng: region?.longitude ?? 0 }}
+                    zoom={10}
+                    mapContainerStyle={containerStyle}
+                    onClick={handleClick}
+                    onIdle={handleIdle}
+                >
+                    {children}
+                </GoogleMap>
+            </LoadScript>
+        );
+    }
 );
 
 export default MapView;

--- a/app-expo/components/MapView.web.tsx
+++ b/app-expo/components/MapView.web.tsx
@@ -1,34 +1,7 @@
 import React from "react";
 import { GoogleMap, Marker as GoogleMarker, LoadScript } from "@react-google-maps/api";
 import type { MapViewProps, MarkerProps } from "./MapView";
-
-// React Native event types used by MapView and Marker
-interface LatLng {
-	latitude: number;
-	longitude: number;
-}
-
-interface Point {
-	x: number;
-	y: number;
-}
-
-interface MarkerPressEvent {
-	nativeEvent: {
-		id: string;
-		action: "marker-press";
-		coordinate: LatLng;
-		position?: Point;
-	};
-}
-
-interface MapPressEvent {
-	nativeEvent: {
-		coordinate: LatLng;
-		position: Point;
-		action?: "marker-press" | "press";
-	};
-}
+import type { MapPressEvent, MarkerPressEvent } from "react-native-maps";
 
 interface RegionChangeDetails {
 	isGesture: boolean;
@@ -38,7 +11,7 @@ export const Marker: React.FC<MarkerProps> = ({ coordinate, title, onPress, test
 	const handleClick = React.useCallback(
 		(e: google.maps.MapMouseEvent) => {
 			if (!onPress || !e.latLng) return;
-			const event: MarkerPressEvent = {
+			const event = {
 				nativeEvent: {
 					id: testID ?? "",
 					action: "marker-press",
@@ -47,7 +20,7 @@ export const Marker: React.FC<MarkerProps> = ({ coordinate, title, onPress, test
 						longitude: e.latLng.lng(),
 					},
 				},
-			};
+			} as unknown as MarkerPressEvent;
 			onPress(event);
 		},
 		[onPress, testID],
@@ -98,7 +71,7 @@ const MapView = React.forwardRef<google.maps.Map | null, MapViewProps>(
 		const handleClick = React.useCallback(
 			(e: google.maps.MapMouseEvent) => {
 				if (!onPress || !e.latLng) return;
-				const pressEvent: MapPressEvent = {
+				const pressEvent = {
 					nativeEvent: {
 						coordinate: {
 							latitude: e.latLng.lat(),
@@ -106,7 +79,7 @@ const MapView = React.forwardRef<google.maps.Map | null, MapViewProps>(
 						},
 						position: { x: 0, y: 0 },
 					},
-				};
+				} as unknown as MapPressEvent;
 				onPress(pressEvent);
 			},
 			[onPress],

